packages/quantic/force-app/main/default/lwc/quanticResultActionBar/quanticResultActionBar.js

```diff
@@ -0,0 +1,76 @@
+import {LightningElement} from 'lwc';
+
+/**
+ *  The `QuanticResultActionBar` component displays the Quantic Result Actions in a proper way with the right styles.
+ * @category Search
+ * @category Insight Panel
+ * @example
+ * <c-quantic-result-action-bar>
+ *   <c-quantic-result-action onclick={actionOne} icon-name="utility:attach" label-when-hover="Action One"></c-quantic-result-action>
+ *   <c-quantic-result-action onclick={actionTwo} icon-name="utility:email" label-when-hover="Action Two"></c-quantic-result-action>
+ *   <c-quantic-result-action onclick={actionThree} icon-name="utility:chat" label-when-hover="Action Three"></c-quantic-result-action>
+ * </c-quantic-result-action-bar>
+ */
+export default class QuanticResultActionBar extends LightningElement {
+  resultActionButtons = [];
+
+  connectedCallback() {
+    this.addEventListener('resultactionregister', this.handleButtonRegister);
+  }
+
+  disconnectedCallback() {
+    this.removeEventListener('resultactionregister', this.handleButtonRegister);
+  }
+
+  /**
+   * Handles the result action button registration event.
+   */
+  handleButtonRegister = (event) => {
+    event.stopPropagation();
+    const callbacks = event.detail;
+    const ref = event.target;
+    this.resultActionButtons.push({callbacks, ref});
+  };
+
+  /**
+   * Handles the slot change event.
+   */
+  handleSlotChange() {
+    this.resultActionButtons = this.getSortedButtons(this.resultActionButtons);
+
+    if (this.resultActionButtons.length === 1) {
+      this.resultActionButtons[0].callbacks.setOrder(null);
+    } else {
+      this.resultActionButtons.forEach((resultActionButton, index, array) => {
+        const order =
+          index === 0
+            ? 'first'
+            : index === array.length - 1
+            ? 'last'
+            : 'middle';
+        resultActionButton.callbacks.setOrder(order);
+      });
+    }
+  }
```

Comment:
If possible, we should try to remove the need for this. I think each button should know their function to call on click, and the order shouldn't matter.

Reply:
agreed

Reply:
to be fair, this doesn’t have anything to do with the function that need to be called on click and the order doesn't matter when it comes to that.
The only thing this order logic is doing, is to make the QuanticResultAction components display in the proper way.
 by proper wayI mean the following:

- when order === 'first' display the button with rounded left corners.
- when order === 'middle' display the button with sharp corners.
- when order === 'last' display the button with rounded right corners.
- when order === null display the button with ALL  rounded corners. this happens when there is only one result action button.

That's what's this logic is doing, the Quantic Result Action  component needs this information about the order, I could've used a property to feed this information to the QuanticResultAction but since the QuanticResultAction already has this information, why not just passing it to the QuanticResultAction without the need to add any new property.

Reply:
I'm convinced there's a better CSS only way to handle that 😂

Reply:
haha x) any idea in mind ?